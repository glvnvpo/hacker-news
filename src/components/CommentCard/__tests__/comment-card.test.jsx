import React from 'react';
import {mount} from 'enzyme';
import {CommentCard} from "../index";

describe('CommentCard', () => {
	let wrapper;

	const comment = {
		id: 123,
		by: 'coolUser',
		text: 'someText',
		time: 1666689031,
		kids: [124],
		showChildComment: true
	};
	const childComment = {
		id: 124,
		by: 'coolUser124',
		text: 'someText124',
		time: 1666689031,
		parent: 123
	};
	const showAnswers = jest.fn();

	const getComponent = (parentComment: Comment) => {
		return mount(
			<CommentCard comment={parentComment} showAnswers={showAnswers}>
				<CommentCard comment={childComment} isParent={false}/>
			</CommentCard>);
	};

	beforeEach(() => {
		jest.clearAllMocks();
	});
	
	it('should render correct card', () => {
		wrapper = getComponent(comment);
		expect(wrapper).toMatchSnapshot();
	});

	it('should render correct text when comment was deleted', () => {
		wrapper = getComponent({...comment, deleted: true, by: undefined});
		expect(wrapper.find('.card-text').at(0).text()).toBe('This comment was deleted');
	});

	it('should render correct text when comment is dead', () => {
		wrapper = getComponent({...comment, dead: true});
		expect(wrapper.find('.card-text').at(0).text()).toBe('Comment not available');
	});

	it('should call showAnswers with parent comment', () => {
		wrapper = getComponent(comment);
		wrapper.find('.comment .parent .btn').simulate('click');
		expect(showAnswers).toHaveBeenCalledWith(comment);
	});

	it('should render correct button text when comment tree is closed', () => {
		wrapper = getComponent({...comment, showChildComment: false});
		const result = wrapper.find('.comment .parent .btn').text();
		expect(result).toBe('Show answers');
	});

	it('should render spinner in button', () => {
		wrapper = getComponent({...comment, isLoadingChildren: true});
		const spinner = wrapper.find('.comment .parent .btn Spinner');
		expect(spinner).toHaveLength(1);
	});
});