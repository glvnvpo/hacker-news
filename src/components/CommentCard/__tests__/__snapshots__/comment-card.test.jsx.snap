// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard should render correct card 1`] = `
<CommentCard
  comment={
    Object {
      "by": "coolUser",
      "id": 123,
      "kids": Array [
        124,
      ],
      "showChildComment": true,
      "text": "someText",
      "time": 1666689031,
    }
  }
  showAnswers={[MockFunction]}
>
  <Card
    body={false}
    border="primary"
    className="comment parent"
  >
    <div
      className="comment parent card border-primary"
    >
      <CardBody>
        <div
          className="card-body"
        >
          <CardSubtitle
            className="mb-2 color-grey"
          >
            <ForwardRef
              className="mb-2 color-grey card-subtitle"
            >
              <div
                className="mb-2 color-grey card-subtitle h6"
              >
                <span
                  className="bold"
                >
                  coolUser
                </span>
                 | 
                25.10.2022
              </div>
            </ForwardRef>
          </CardSubtitle>
          <CardText
            as="span"
            className="bold color-dark-grey"
          >
            <span
              className="bold color-dark-grey card-text"
            >
              someText
            </span>
          </CardText>
          <Button
            active={false}
            className="mt-20"
            disabled={false}
            onClick={[Function]}
            variant="light"
          >
            <button
              className="mt-20 btn btn-light"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              Hide answers
            </button>
          </Button>
          <CommentCard
            comment={
              Object {
                "by": "coolUser124",
                "id": 124,
                "parent": 123,
                "text": "someText124",
                "time": 1666689031,
              }
            }
            isParent={false}
          >
            <Card
              body={false}
              border={false}
              className="comment child"
            >
              <div
                className="comment child card"
              >
                <CardBody>
                  <div
                    className="card-body"
                  >
                    <CardSubtitle
                      className="mb-2 color-grey"
                    >
                      <ForwardRef
                        className="mb-2 color-grey card-subtitle"
                      >
                        <div
                          className="mb-2 color-grey card-subtitle h6"
                        >
                          <span
                            className="bold"
                          >
                            coolUser124
                          </span>
                           | 
                          25.10.2022
                        </div>
                      </ForwardRef>
                    </CardSubtitle>
                    <CardText
                      as="span"
                      className="bold color-dark-grey"
                    >
                      <span
                        className="bold color-dark-grey card-text"
                      >
                        someText124
                      </span>
                    </CardText>
                  </div>
                </CardBody>
              </div>
            </Card>
          </CommentCard>
        </div>
      </CardBody>
    </div>
  </Card>
</CommentCard>
`;
